draft-ietf-lake-reqs.md

```diff
@@ -247,20 +247,28 @@ However, PSK-based provisioning has inherent weaknesses. There has been reports
 Shared keys can alternatively be established in the endpoints using an AKE protocol authenticated with asymmetric public keys instead of symmetric secret keys. Raw public keys (RPK) can be provisioned with the same scheme as PSKs, which allows for a more relaxed trust model since RPKs need not be secret. The corresponding private keys are assumed to be provisioned to the party being authenticated beforehand (e.g. in factory or generated on-board).
 
 As a third option, by using a public key infrastructure and running an asymmetric key AKE with public key certificates instead of RPKs, key provisioning can be omitted, leading to a more automated ("zero-touch") bootstrapping procedure. The root CA keys are assumed to be provisioned beforehand.
+Public key certificates are important for several IoT settings, e.g., facility management with a large number of devices from many different manufacturers.
 
 These steps provide an example of a migration path in limited scoped steps from simple to more robust security bootstrapping and provisioning schemes where each step improves the overall security and/or simplicity of deployment of the IoT system, although not all steps are necessarily feasible for the most constrained settings.
 
 In order to allow for these different schemes, the AKE must support PSK- (shared between two nodes), RPK- and certificate-based authentication.
+These are also the schemes for which CoAP is designed (see Section 9 of {{RFC7252}}).
 
 Multiple public key authentication credential types may need to be supported for RPK and certificate-based authentication. In case of a Diffie-Hellman key exchange both the use of signature based public keys (for compatibility with existing ecosystem) and static DH public keys (for reduced message size) is expected.
 
-To further minimize the bandwidth consumption it is required to support transporting certificates and raw public keys by reference rather than by value. Considering the wide variety of deployments, the AKE must support different schemes for transporting and identifying credentials, including x5t, x5u and x5chain (see Section 2 of {{I-D.ietf-cose-x509}}).
+To further minimize the bandwidth consumption it is required to support transporting certificates and raw public keys by reference rather than by value. Considering the wide variety of deployments, the AKE must support different schemes for transporting and identifying credentials, including certificate by hash, certificate by URL, and certificates explicitly, such as x5t, x5u and x5chain (see Section 2 of {{I-D.ietf-cose-x509}}).
```

Comment:
My understanding of Ekr's concerns leads me to think that he will not  find this text satisfactory (and did not find the previous text satisfactory, for that matter).  Something like "[...] must support different schemes for transporting and identifying credentials.  While there are many existing mechanisms for doing so, ranging from PSK or raw public key by reference all the way through a full x5chain of in-band certificates, what is appropriate for a given deployment will depend on the nature of that deployment.  In order to provide a clear initial effort, {{initial-focus}} lists a set of credential types of immediate relevance; the mechanism for a selecting credentials scheme is presumed to enable future extensibility if needed."

Reply:
This change was based on Ekr's comment in mail to the LAKE WG mailing list dated March 31, the same day as we had the virtual interim meeting. 

I clarified according to your proposal, see e97e434.

Reply:
First, a auestion: you are proposing to remove PSK? What about resumption?

Second, it's not really a matter of satisfactory or not satisfactory. The requirements are what they are. However, it's not clear to me that this is a meaningful reduction in scope.

Reply:
I argued for the initial focus in this mail:
https://mailarchive.ietf.org/arch/msg/lake/wT9kO9hGWRx_9V3kz36lHTKx0h8/
Please let us know if anything of that was not clear.

If RPK ECHDE is a similar cost compared to PSK ECDHE then PSK ECHDE is less urgent and can be addressed later. There are also resumption mechanisms for OSCORE, existing and additional as discussed on the list.

Reply:
I don't believe that OSCORE resumption (assuming you are referring to https://tools.ietf.org/html/rfc8613#appendix-B.2) provides the same FS properties as resumption with DH. Perhaps that's unnecessary, but the consequence of this, I believe, is that if you want FS you will need to do a full handshake.

Reply:
This is one example from the discussion on the list I was referring to:
https://mailarchive.ietf.org/arch/msg/lake/5DAmN2c7cOXCUej-4XRHxa18MIQ/